test-framework/common/src/main/java/io/quarkus/test/common/ProcessReader.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.test.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+class ProcessReader implements Runnable {
+
+    private final InputStream inputStream;
+
+    ProcessReader(InputStream inputStream) {
+        this.inputStream = inputStream;
+    }
+
+    @Override
+    public void run() {
+        handleStart();
+        byte[] b = new byte[100];
+        int i;
+        try {
+            while ((i = inputStream.read(b)) > 0) {
+                String str = new String(b, 0, i, StandardCharsets.UTF_8);
+                System.out.print(str);
```

Comment:
Its probably a massive corner case that we don't care about, but this won't display properly if you have a muti byte character that falls on the 100 byte mark.

Reply:
Sounds like a corner case indeed 🙂.
But I'll keep it in mind and deal with it if it ever becomes a problem.